coilsnake/model/eb/palettes.py

```diff
@@ -244,12 +244,21 @@ def get_color_id(self, rgb, subpalette_id):
 
     def __getitem__(self, key):
         subpalette_number, color_number = key
-        if subpalette_number < 0 or color_number < 0 or subpalette_number >= self.num_subpalettes \
-                or color_number >= self.subpalette_length:
+        if subpalette_number < 0 or subpalette_number >= self.num_subpalettes \
+                or (isinstance(color_number, int) and (color_number < 0 or color_number >= self.subpalette_length)):
```

Comment:
Yes because I use slices in my code too. I initially added additional checks on the slices too, but it became too complicated to handle all the cases (positive, negative, None, etc.). I could add those checks back if you think it would be useful, or I could not use slices. Without slices though, the code would be a lot heavier in certain parts.